### DISCUSSION TITLE
リリースの頻度について

### DISCUSSION THREAD

**Discussion by tokuhirom** [akaza-im/akaza#171] [General]
リリースの頻度は当面の間は、何かユーザー側に影響あることがあったタイミングでマメにやる。
わりとgithub actionsのクオーター（ディスク容量、CPU時間など）のギリギリになりがちなので、大規模な変更を加えたあげくにリリースできない、ということがありうる。
また、リリース作業にモデルの構築が含まれるので2時間はかかる。

開発が落ち着いてきたら29の日リリースぐらいの感じになっていくのだと思う。

**Comment by omasanori** [comment]
モデルのフォーマットが安定したら、モデルのリリースとコードのリリースを分離するのはどうでしょうか。極端な話、akaza-dataのリポジトリを分けるとか。

**Reply by tokuhirom** [reply]
それもアリだと思います。

**Reply by tokuhirom** [reply]
いずれにせよ複数のレポジトリに分けるタイミングは来ると思うので、多分、そのタイミングで github org を作る感じかな〜と

**Reply by tokuhirom** [reply]
最終的には、bccwjとかなんらかのクローリングデータを元にしたモデルを第三者がリリース出来るようになって欲しいと考えると、今のakaza-dataのsubcmd でハードコードになってる部分をパラメータとして外部から設定できるようにするなどの処置が必要。

**Reply by tokuhirom** [reply]
一旦、org に transfer しました

**Reply by tokuhirom** [reply]
https://github.com/akaza-im/akaza-default-model モデル作成を完全に別レポジトリに移しました。
デフォルトのモデルといえども、完全に別になっているほうが、管理上楽そうなのと、別管理になっているほうがモデルの差し替えの心理的障壁が下がるかなと。

**Reply by tokuhirom** [reply]
分離しようとしていますが、一時的に構成がグッチャグチャになっています。

**Reply by tokuhirom** [reply]
akaza-default-model の側の CI も make evaluate が後一歩で通るというところまできた。

**Reply by tokuhirom** [reply]
対応完了!

**Comment by omasanori** [comment]
話は変わりますが、現状`Cargo.toml`とタグのバージョンが一致していません。[release-plz](https://crates.io/crates/release-plz)かなにかでいい感じにするとよさそう。

**Reply by omasanori** [reply]
もしくは https://github.com/cargo-bins/release-pr でリリース作業をGitHub Action化するとか。

**Reply by tokuhirom** [reply]
git tag を打つとリリースされるようになってはいるんですよね。
ただ、バージョンの書き換えをサボっているという。。
それ系やりたいところではあります

**Reply by tokuhirom** [reply]
release-plz は試してみたけどうまく動かないですね。workspace になっていて複数のプロジェクトが変更されるケースには対応していない様子。

**Reply by tokuhirom** [reply]
`TAG=0.1.7 perl -i -pe 's/^(version = ").*"/$1$ENV{TAG}"/' */Cargo.toml`

本格的になにかするのはともかくとして、今はこんなんでいいかな。。

**Comment by tokuhirom** [comment]
それは薄々感づいてまして、やった方がいいな、と思ってるんですが、手が回ってません。。😭
